### PR TITLE
Issue 28620 show preview button does not reflect selected language

### LINKED ISSUE #28620: Issue: "Show Preview" Button Does Not Reflect Selected Language
### Parent Issue

_No response_

### Problem Statement

When switching the language in Edit Mode, the Preview updates correctly to the selected language. However, clicking the "Show Preview" button opens a new tab with the content in the default language (English) instead of the selected language (Spanish).

### Steps to Reproduce

Steps to Reproduce

1. Open a page in Edit Mode in English.
2. Use the language dropdown on the right to switch to Spanish.
3. Observe that the Preview updates to Spanish.
4. Click the "Show Preview" button.

Expected Behavior
The new tab opened by clicking the "Show Preview" button should display the page content in the selected language (Spanish).

Actual Behavior
The new tab opened by clicking the "Show Preview" button displays the page content in the default language (English), despite the Edit Mode showing the correct language (Spanish).

Technical Details
The URL in Edit Mode updates to language_id=2 when switching to Spanish.
The URL used by the "Show Preview" button remains language_id=1, indicating English.

Uploading Screen Recording 2024-05-21 at 10.56.03 AM.mov…

### Acceptance Criteria

Language Consistency in Preview:

When a user switches the language in Edit Mode using the language dropdown, the Preview section should update to the selected language.
Clicking the "Show Preview" button should open a new tab that reflects the language selected in Edit Mode.

### dotCMS Version

23.10.24 LTS

### Proposed Objective

Core Features

### Proposed Priority

Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

https://dotcms.zendesk.com/agent/tickets/116371

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by gortiz-dotcms**
Added a missing set to apiLink variable when changing values on edit page tab, this way when changing language from the dropdown and clicking "Show preview" button, it is going to have a valid href with correct language value so it can redirect to the correct preview

This PR fixes: #28620

### CODE REVIEW

**Review by erickgonzalez — APPROVED** [review]
Looks good to me. Gonna ask review of some of the FE devs

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/seo/components/dot-edit-page-toolbar-seo/dot-edit-page-toolbar-seo.component.ts [hunk 1]
```diff
@@ -111,7 +111,7 @@ export class DotEditPageToolbarSeoComponent implements OnInit, OnChanges, OnDest
 
     ngOnChanges(): void {
         this.pageRenderedHtml = this.updateRenderedHtml();
-
+        this.apiLink = `api/v1/page/render${this.pageState.page.pageURI}?language_id=${this.pageState.page.languageId}`;
```
Comment (oidacra) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/seo/components/dot-edit-page-toolbar-seo/dot-edit-page-toolbar-seo.component.ts:114]: Remember to add a test for this value.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/seo/components/dot-edit-page-toolbar-seo/dot-edit-page-toolbar-seo.component.spec.ts [hunk 2]
```diff
@@ -481,4 +481,14 @@ describe('DotEditPageToolbarSeoComponent', () => {
             });
         });
     });
+
+    describe('ngOnChange', () => {
+        it('should have a new api link', () => {
+            component.apiLink = '';
+            const expectedLink = `api/v1/page/render${componentHost.pageState.page.pageURI}?language_id=${componentHost.pageState.page.languageId}`;
+            fixtureHost.detectChanges();
+            component.ngOnChanges();
```
Comment (oidacra) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/seo/components/dot-edit-page-toolbar-seo/dot-edit-page-toolbar-seo.component.spec.ts:490]: Pass the responsibility to the component to trigger `onChanges`; if any input of the component changes, it should be triggered.
```
component.setInput('pageState',mockValue)
fixtureHost.detectChanges();

expect(component.apiLink).toBe(expectedLink);
```

**Review by hmoreras — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/seo/components/dot-edit-page-toolbar-seo/dot-edit-page-toolbar-seo.component.spec.ts [hunk 3]
```diff
@@ -481,4 +482,33 @@ describe('DotEditPageToolbarSeoComponent', () => {
             });
         });
     });
+
+    describe('ngOnChange', () => {
+        beforeEach(() => {
+            de = deHost.query(By.css('dot-edit-page-view-as-controller'));
+        });
+
+        xit('should have a new api link', async () => {
```
Comment (oidacra) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/seo/components/dot-edit-page-toolbar-seo/dot-edit-page-toolbar-seo.component.spec.ts:491]: Why skip?

Comment (gortiz-dotcms) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/seo/components/dot-edit-page-toolbar-seo/dot-edit-page-toolbar-seo.component.spec.ts:491]: my bad already changing it

**Review by oidacra — APPROVED** [review]

**Pull request merged**